### PR TITLE
Profile Playlists and viewing other peoples' profiles

### PR DESCRIPTION
**Pull request by jack-michaud**
- Adds /profile/:profileId route on the frontend 
  - Fetches profile information for a given profile Id. We use the same profile detail component, we just push a different profile fetched from /users/:userId to it
- Surfaces playlist information through /me and /users/:userId. 

image
image

### CODE REVIEW

File: backend_src/http_api/auth.js [hunk 1]
```diff
@@ -20,20 +20,25 @@ const getToken = async () => {
   return access_token;
 };
 
-const refreshAccessToken = async (refresh_token_old) => {
-  const { data: { access_token, refresh_token } } = await axios({
```
Comment (MichaelBond24) [inline backend_src/http_api/auth.js:24]: Looks like this explains how I ended up with a null refresh_token before. Do refresh tokens never expire?

File: backend_src/http_api/routes/users.js [hunk 2]
```diff
@@ -20,49 +20,67 @@ router.get('/me', async (req, res) => {
...
-        data.playlists = data.playlists.items.concat(sharedPlaylists);
-      }
+
+const getPlaylists = async ({ user, private }) => {
+  if (private == null) {
+    private = false;
+  }
+  
+  if (private) {
+    const { access_token, refresh_token } = await auth.refreshAccessToken(user.refreshToken);
```
Comment (MichaelBond24) [inline backend_src/http_api/routes/users.js:40]: This refresh_token is the same one that was passed in right? So we don't need to update the DB with it. But that change is fine to make later

File: backend_src/http_api/auth.js [hunk 3]
```diff
@@ -20,20 +20,25 @@ const getToken = async () => {
...
+      headers: {
+        'Authorization': 'Basic ' + Buffer.from(CLIENT_ID + ':' + CLIENT_SECRET).toString('base64'),
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      params: {
+        grant_type: 'refresh_token',
+        refresh_token
+      },
+    });
+    return { access_token, refresh_token };
```
Comment (MichaelBond24) [inline backend_src/http_api/auth.js:37]: Don't need to return refresh_token if it's just the same one that was passed in, but we can change that later

File: backend_src/http_api/routes/users.js [hunk 4]
```diff
@@ -20,49 +20,67 @@ router.get('/me', async (req, res) => {
...
+    }
   }
+}
+router.get('/:userId/playlists', async (req, res) => {
+  const { userId } = req.params;
+  const user = await db.get().collection('users').findOne({ _id: userId });
+
+  const private = req.session.spotifyId === userId;
+
+  return getPlaylists({ user, userId });
```
Comment (MichaelBond24) [inline backend_src/http_api/routes/users.js:83]: I think you meant to pass in `private` as the second parameter here, not `userId`

File: backend_src/http_api/auth.js (hunk 1 above)
Comment (jack-michaud) [reply backend_src/http_api/auth.js:24]: Yeah, I guess not. I assumed incorrectly that we'd be getting a new one each time D:

File: backend_src/http_api/auth.js (hunk 3 above)
Comment (jack-michaud) [reply backend_src/http_api/auth.js:37]: Yeah, I left it to maintain back-compat/laziness

File: backend_src/http_api/routes/users.js (hunk 4 above)
Comment (jack-michaud) [reply backend_src/http_api/routes/users.js:83]: I think you're super right

Comment (jack-michaud) [reply backend_src/http_api/routes/users.js:83]: It managed to work anyways because if private is null it defaults to false hah

**Pull request merged**